### DISCUSSION TITLE
Timeseries modeling from a pandas dataframe

### DISCUSSION THREAD

**Discussion by aropele** [fastapi/sqlmodel#639] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import pandas as pd
from sqlmodel import Field, Session, SQLModel, create_engine
from datetime import datetime


class StockTimeSeries(SQLModel, table=True):
    id: int = Field(primary_key=True)
    symbol: str
    timestamp: datetime
    price: float


sqlite_file_name = "stock_database.db"
sqlite_url = f"sqlite:///{sqlite_file_name}"
engine = create_engine(sqlite_url, echo=True)


def create_db_and_tables():
    SQLModel.metadata.create_all(engine)


def insert_dataframe_to_db(df):
    with Session(engine) as session:
        for _, row in df.iterrows():
            time_series_entry = StockTimeSeries(
                id=row["id"],
                symbol=row["symbol"],
                timestamp=row["timestamp"],
                price=row["price"],
            )
            session.add(time_series_entry)

        session.commit()


def main():
    create_db_and_tables()

    # Simulating a Pandas DataFrame for demonstration
    data = {
        "id": [1, 2, 3],
        "symbol": ["AAPL", "AAPL", "AAPL"],
        "timestamp": [
            datetime(2023, 8, 1, 9, 0),
            datetime(2023, 8, 1, 10, 0),
            datetime(2023, 8, 1, 11, 0),
        ],
        "price": [150.0, 155.0, 160.0],
    }
    df = pd.DataFrame(data)

    insert_dataframe_to_db(df)


if __name__ == "__main__":
    main()
```

### Description

Hi, I just started using sqlmodel and I would like to know if this way of managing time series is the optimal one.
In particular I need to write timeseries that are stored in side a pandas df. So:
- I iterate over the dataframe and trasforming each row to a StockTimeSeries instance, before writing each instance
- the class StockTimeSeries is modeling a single time series row, not an entire series

Is this correct or there are some other way to solve this requirement?

Thanks.

### Operating System

Windows

### SQLModel Version

0.0.8

### Python Version

3.9

**Comment by YuriiMotov** [comment, marked as answer]
Your approach looks good.

For big data volume, you can use bulk insert:
```py
data = [
    {
        "id": row["id"],
        "symbol": row["symbol"],
        ...
    }
    for _, row in df.iterrows()
]

session.exec(StockTimeSeries, params=data)
session.commit()
```
